app/src/main/java/co/develhope/chooseyouowncocktail_g2/DetailDrinkPageFragment.kt

```diff
@@ -44,34 +46,30 @@ class DetailDrinkFragment : Fragment() {
         return binding.root
     }
 
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.title.text = param_name
-        binding.description.text = param_desc
-
-        Picasso.get()
-            .load("https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg")
-            .resize(200,200)
-            .placeholder(R.drawable.placeholder)
-            .error(R.drawable.placeholder)
-            .into(binding.preview)
-
-        binding.cl.text = param_cl
-
-        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
-            when (param_currentPage) {
-                "Search" -> view.findNavController()
-                    .navigate(R.id.action_detailDrinkFragment_to_navigation_search)
-                "Home" -> view.findNavController()
-                    .navigate(R.id.action_detailDrinkFragment_to_navigation_home)
+        arguments?.let {
+            DrinkList.getByID(it.getInt(param_drink_ID)).let {
+                binding.title.text = it!!.name
```

Comment:
vi ho rispetuto più volte che Il double bang (!!) non lo prendo in considerazione come soluzione, bisogna gestire la possibilità di null in modo corretto.
es. se non è null prende name e se dovesse essere null che succede? state usando let usatelo per controllare la nullability dell'oggetto che gli passate dentro. Se per caso fosse null potreste mostrare una textview con un messaggio di errore